framework/freedomotic-core/src/test/java/com/freedomotic/rules/StatementTest.java

```diff
@@ -0,0 +1,108 @@
+package com.freedomotic.rules;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class StatementTest {
+
+  @Test
+  public void createLogicalInputValid() {
+    final Statement statement = new Statement();
+    final Statement retval = statement.create(Statement.EQUALS, "retval1", Statement.AND, "100");
+    Assert.assertNotNull(retval);
+    Assert.assertEquals(Statement.EQUALS, retval.getLogical());
+    Assert.assertEquals("retval1", retval.getAttribute());
+    Assert.assertEquals(Statement.AND, retval.getOperand());
+    Assert.assertEquals("100", retval.getValue());
+  }
+
+  @Test
+  public void createLogicalInputInvalid() {
+    final Statement statement = new Statement();
+    final Statement retval = statement.create(Statement.NOT, "retval2", Statement.OR, "100");
+    Assert.assertNotNull(retval);
+    Assert.assertEquals(Statement.NOT, retval.getLogical());
+    Assert.assertEquals("retval2", retval.getAttribute());
+    Assert.assertEquals(Statement.OR, retval.getOperand());
+    Assert.assertEquals("100", retval.getValue());
+  }
+
+  @Test
+  public void createStatementNull() {
+    Assert.assertNull(new Statement().create(" ", null, " ", null));
+  }
+
+  @Test
+  public void statementEqualsTrue() {
+    Assert.assertEquals(new Statement().create(null, "!", null, " "),
+            new Statement().create(null, "!", "!", null));
+  }
+
+  @Test
+  public void statementEqualsFalse() {
+    Assert.assertNotEquals(new Statement().create(" ", null, " ", null),
+            new Statement().create(Statement.LESS_THAN, "retval3", Statement.OR, "50"));
+  }
+
+  @Test
+  public void equalsNullNotEqual() {
+    Assert.assertNotEquals(new Statement(), " ");
+  }
+
+  @Test
+  public void getValueEqualsEmptyString() {
+    Assert.assertEquals("", new Statement().getValue());
+  }
+
+  @Test
+  public void getValueNonEmptyString() {
+    Assert.assertEquals("14", new Statement().create(Statement.LESS_THAN, "retval1", Statement.OR, "14").getValue());
+  }
+
+  @Test
+  public void equalsLogicalFalse() {
+    Assert.assertNotEquals(new Statement().create(Statement.GREATER_EQUAL_THAN, "retval1", Statement.NOT, "10"),
+            new Statement().create(Statement.LESS_THAN, "retval1", Statement.NOT, "10"));
+  }
+
+  @Test
+  public void equalsAttributeFalse() {
+    Assert.assertNotEquals(new Statement().create(Statement.GREATER_EQUAL_THAN, "retval1", Statement.NOT, "10"),
+            new Statement().create(Statement.GREATER_EQUAL_THAN, "retval2", Statement.NOT, "10"));
+  }
+
+  @Test
+  public void equalsOperandFalse() {
+    Assert.assertNotEquals(new Statement().create(Statement.GREATER_EQUAL_THAN, "retval1", Statement.NOT, "10"),
+            new Statement().create(Statement.GREATER_EQUAL_THAN, "retval1", Statement.OR, "10"));
```

Comment:
Might be worth using indentation that lines up the two create calls, so that their differences can be seen more easily.